includes/admin/class-sensei-plugins-installation.php

```diff
@@ -0,0 +1,421 @@
+		set_transient( self::INSTALLING_PLUGINS_TRANSIENT, $installing_plugins, DAY_IN_SECONDS );
+	}
+
+	/**
+	 * Get file name from path.
+	 *
+	 * @param string $path Complete path.
+	 *
+	 * @return string File name.
+	 */
+	private function get_file_name_from_path( $path ) {
+		$path_array = explode( '/', $path );
+		return end( $path_array );
+	}
+
+	/**
+	 * Get slug from path and associate it with the path.
+	 *
+	 * @param array  $plugins Associative array of plugin files to paths.
+	 * @param string $key     Plugin relative path. Example: woocommerce/woocommerce.php.
+	 */
+	private function associate_plugin_file( $plugins, $key ) {
+		$filename             = $this->get_file_name_from_path( $key );
+		$plugins[ $filename ] = $key;
+		return $plugins;
+	}
+
+	/**
+	 * Get installed plugins.
+	 *
+	 * @return array Installed plugins.
+	 */
+	private function get_installed_plugins() {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		return array_reduce( array_keys( get_plugins() ), [ $this, 'associate_plugin_file' ] );
+	}
+
+	/**
+	 * Check whether plugin is active.
+	 *
+	 * @param string $plugin_file       Plugin filename or path.
+	 * @param array  $installed_plugins Installed plugins. If it's not passed, it will get the installed plugins.
+	 *
+	 * @return boolean Is plugin active.
+	 */
+	public function is_plugin_active( $plugin_file, $installed_plugins = null ) {
+		if ( ! $installed_plugins ) {
+			$installed_plugins = $this->get_installed_plugins();
+		}
+
+		// Get filename if it is the complete path.
+		$plugin_file = $this->get_file_name_from_path( $plugin_file );
+
+		if ( ! isset( $installed_plugins[ $plugin_file ] ) ) {
+			return false;
+		}
+
+		return is_plugin_active( $installed_plugins[ $plugin_file ] );
+	}
+
+	/**
+	 * Save error
+	 *
+	 * @param string $slug    Plugin slug.
+	 * @param string $message Error message.
+	 */
+	private function save_error( $slug, $message ) {
+		$installing_plugins = $this->get_installing_plugins();
+		$key                = array_search( $slug, array_column( $installing_plugins, 'product_slug' ), true );
+
+		if ( false !== $key ) {
+			$installing_plugins[ $key ]->error = $message;
+		}
+
+		$this->set_installing_plugins( $installing_plugins );
+	}
+
+	/**
+	 * Complete installation removing the plugin from the transient.
+	 *
+	 * @param string $plugin_slug
+	 */
+	private function complete_installation( $plugin_slug ) {
+		$installing_plugins = $this->get_installing_plugins();
+
+		if ( ! empty( $installing_plugins ) ) {
+			$installing_plugins = array_filter(
+				$installing_plugins,
+				function( $plugin ) use ( $plugin_slug ) {
+					return $plugin->product_slug !== $plugin_slug;
+				}
+			);
+
+			$this->set_installing_plugins( $installing_plugins );
+		}
+	}
+
+	/**
+	 * Wrapper to get error message and give the `get_error_data` as fallback.
+	 *
+	 * @param WP_Error $error
+	 *
+	 * @return string Error message.
+	 */
+	private function get_error_message( $error ) {
```

Comment:
I created this wrapper because `get_error_message` was coming empty for some cases. And this cases had a message in the `error_data`.